pacman/model/partitioner_interfaces/abstract_splitter_partitioner.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2020-2021 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from six import add_metaclass
+from spinn_utilities.abstract_base import AbstractBase, abstractmethod
+
+
+@add_metaclass(AbstractBase)
+class AbstractSplitterPartitioner(object):
+    """ Splitter API to allow other Partitioner's to add more stuff to the
+    edge creation process.
+
+    This makes sure that the methods super class expect to be there are not
+    removed.
+
+    """
+
+    @abstractmethod
+    def create_machine_edge(
+            self, src_machine_vertex, dest_machine_vertex,
+            common_edge_type, app_edge, machine_graph,
+            app_outgoing_edge_partition, resource_tracker):
+        """ Creates the machine edge (if needed) and adding it
+        to the graph
+
+        Some implementations of this method are able to detect that the
+        requested edge is not actually needed so never create or add it.
+
+        :param MachineVertex src_machine_vertex: Src machine vertex of a edge
```

Comment:
In general, because this is a method that is supposed to support `@overrides` (and that's pretty stupid, as is Sphinx), the types of the arguments should be the fully qualified forms with a preceding tilde (e.g., `~pacman.model.graphs.machine.MachineVertex` on this line).